pytest_picked/plugin.py

```diff
@@ -53,8 +53,7 @@ def _get_affected_paths(config):
         _write(config, [error])
         config.args = []
         raise ValueError(error)
-    else:
-        return mode.affected_tests()
+    return mode[0](*mode[1:]).affected_tests()
```

Comment:
What about changing it to something more readable?

```
    modes = {
        "branch": {
            "class": Branch,
            "args": [test_file_convention, parent_branch]
        },
        "unstaged": {
            "class": Unstaged,
            "args": [test_file_convention]
        },
    }
    try:
        mode = modes[picked_mode]
    except KeyError:
        error = "Invalid mode. Options: `{}`.".format(", ".join(modes.keys()))
        _write(config, [error])
        config.args = []
        raise ValueError(error)
    return mode["class"](*mode["args"]).affected_tests()
```

It is more verbose, but it helps when reading the code.

Reply:
I also had a version briefly where I used `partial` to pass `Branch` its extra kwarg, if you'd prefer that.

Reply:
Alternatively, maybe lets pass `test_file_convention` explicitly, and then pass all options as kwargs to all modes and it's up to the mode to make use of whichever things it supports?